### PR TITLE
Graph Footer: Cache State History

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

Cache the state history of the entity and only fetch from the last update until the time of the new update.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/header-footer/hui-graph-header-footer.ts [hunk 1]
```diff
@@ -80,21 +85,54 @@ export class HuiGraphHeaderFooter extends LitElement
...
 
     if (changedProps.has("_config")) {
+      const oldConfig = changedProps.get("_config") as GraphHeaderFooterConfig;
+      if (!oldConfig || oldConfig.entity !== this._config.entity) {
+        this._stateHistory = [];
+      }
+
       this._getCoordinates();
     } else if (Date.now() - this._date!.getTime() >= MINUTE) {
       this._getCoordinates();
```
Comment (bramkragten) [inline src/panels/lovelace/header-footer/hui-graph-header-footer.ts:95]: We should make sure we don't call this while it is still running a fetch

File: src/panels/lovelace/header-footer/hui-graph-header-footer.ts [hunk 2]
```diff
@@ -70,31 +76,66 @@ export class HuiGraphHeaderFooter extends LitElement
...
 
-  protected firstUpdated(): void {
-    this._date = new Date();
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    return hasConfigOrEntityChanged(this, changedProps);
   }
 
   protected updated(changedProps: PropertyValues) {
-    if (!this._config || !this.hass) {
+    if (!this._config || !this.hass || this._fetching) {
```
Comment (bramkragten) [inline src/panels/lovelace/header-footer/hui-graph-header-footer.ts:84]: If the config changes, we should refetch I think

File: src/panels/lovelace/header-footer/hui-graph-header-footer.ts [hunk 3]
```diff
@@ -70,31 +76,66 @@ export class HuiGraphHeaderFooter extends LitElement
...
+        : this._date;
+
+    if (this._stateHistory!.length) {
+      this._stateHistory = this._stateHistory!.filter(
+        (entity) =>
+          endTime.getTime() - new Date(entity.last_changed).getTime() <= DAY
+      );
+    }
+
+    const stateHistory = await fetchRecent(
```
Comment (bramkragten) [inline src/panels/lovelace/header-footer/hui-graph-header-footer.ts:119]: We should `skipInitialState` if it is not the first time we fetch the data

**Comment by bramkragten** [comment]
This 

<https://github.com/home-assistant/frontend/pull/5499/commits/43dc43d9fc1df473c7c112ae0284a4491c2bb19c#diff-4617a1a8ea3c2e00061a4a28c739de4fR98>

Should be inside the check on line 95

**Review by bramkragten — APPROVED** [review]

**Comment by zsarnett** [comment]
I will resolve these today. and merge

**Comment by zsarnett** [comment]
> This
> 
> [43dc43d#diff-4617a1a8ea3c2e00061a4a28c739de4fR98](https://github.com/home-assistant/frontend/commit/43dc43d9fc1df473c7c112ae0284a4491c2bb19c#diff-4617a1a8ea3c2e00061a4a28c739de4fR98)
> 
> Should be inside the check on line 95

@bramkragten I am not 100% what you mean by this. Guessing you are talking about the `this._fetching` but not sure

**Comment by bramkragten** [comment]
> > This
> > [43dc43d#diff-4617a1a8ea3c2e00061a4a28c739de4fR98](https://github.com/home-assistant/frontend/commit/43dc43d9fc1df473c7c112ae0284a4491c2bb19c#diff-4617a1a8ea3c2e00061a4a28c739de4fR98)
> > Should be inside the check on line 95
> 
> @bramkragten I am not 100% what you mean by this. Guessing you are talking about the `this._fetching` but not sure

Line 98 should be in the if check after line 95

image

**Pull request merged**